common/kafka/kafka_consumer.cpp

```diff
@@ -71,30 +73,106 @@ bool KafkaSeekWithRetry(RdKafka::KafkaConsumer* consumer,
+                   << "required conf couldn't be set "
+                   << "key=" << key
+                   << ", value=" << value.first << std::endl;
+        common::Stats::get()->Incr(getFullStatsName(
+          kKafkaConsumerErrorInit, {"kafka_consumer_type=" + kafka_consumer_type}));
+        return nullptr;
+      } else {
+        LOG(ERROR) << "Couldn't set provided config key-value pair"
+                   << "key=" << key
+                   << ", value=" << value.first
+                   << ", error:" << err << std::endl;
+      }
+    }
+  }
+
+  return std::shared_ptr<RdKafka::KafkaConsumer>(
+    RdKafka::KafkaConsumer::create(conf.get(), err));
+}
+
 std::shared_ptr<RdKafka::KafkaConsumer> CreateRdKafkaConsumer(
     const std::unordered_set<uint32_t>& partition_ids,
     const std::string& broker_list,
     const std::string& group_id,
     const std::string kafka_consumer_type) {
   std::string err;
   auto conf = std::shared_ptr<RdKafka::Conf>(RdKafka::Conf::create(
-      RdKafka::Conf::CONF_GLOBAL));
-  // https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
-  if (conf->set("metadata.broker.list", broker_list, err) !=
-  RdKafka::Conf::CONF_OK) {
-    LOG(ERROR) << "Failed to create kafka config for partitions: "
-               << folly::join(",", partition_ids) << ", error: " << err;
-    common::Stats::get()->Incr(getFullStatsName(
-        kKafkaConsumerErrorInit, {"kafka_consumer_type=" + kafka_consumer_type}));
-    return nullptr;
+    RdKafka::Conf::CONF_GLOBAL));
+
+  KafkaConfigMap kafkaConfigMap;
+
+  // Note: Following settings are overridden from configuration.
+  // We explicitly set the eof this flag so that control passes back to
+  // client code one's partition eof is hit. This is important for rocksplicator's
+  // BootstrapStateModel. Since this property default changed in librdkafka-1.0.0-RC4,
+  // from true to false, upgrade to newer version of librdkafka will be impossible
+  // without explicitly setting this parameter to true. However for compatibility
+  // with older versions of librdkafka that may not have
+  // this flag in first place, we don't require this setting to be compatible.
+  // We also allow this setting to be overriden from configuration file.
+  kafkaConfigMap["enable.partition.eof"] = std::make_pair("true", false);
+
+  // This preserves previous behavior. However this setting can be overwritten
+  // from configuration file as well, in which case this need not be set from
+  // flags.
+  kafkaConfigMap["enable.auto.offset.store"] = std::make_pair(
+    FLAGS_enable_kafka_auto_offset_store, false);
+
+  kafka::ConfigMap configMap;
+  if (!FLAGS_kafka_client_global_config_file.empty()) {
+    if (!kafka::KafkaConfig::read_conf_file(
+      FLAGS_kafka_client_global_config_file, &configMap)) {
+      LOG(ERROR) << "Can not read / parse config file: "
+                 << FLAGS_kafka_client_global_config_file
+                 << std::endl;
```

Comment:
Can you also increment the counter for for ErrorInit?

Reply:
common::Stats::get()->Incr(getFullStatsName(
          kKafkaConsumerErrorInit, {"kafka_consumer_type=" + kafka_consumer_type}));
?? You meant stats?

Reply:
Yes, stats counter :)